frb_codegen/src/generator/dart/ty_opaque.rs

```diff
@@ -0,0 +1,54 @@
+use crate::generator::dart::ty::*;
+use crate::ir::*;
+use crate::target::Acc;
+use crate::type_dart_generator_struct;
+
+type_dart_generator_struct!(TypeOpaqueGenerator, IrTypeOpaque);
+
+impl TypeDartGeneratorTrait for TypeOpaqueGenerator<'_> {
+    fn api2wire_body(&self) -> Acc<Option<String>> {
+        Acc {
+            io: Some(format!(
+                "final ptr = inner.new_{0}();
+                _api_fill_to_wire_{0}(raw, ptr);
+                return ptr;",
+                self.ir.safe_ident(),
+            )),
+            wasm: Some("return raw.share();".to_owned()),
+            ..Default::default()
+        }
+    }
+
+    fn api_fill_to_wire_body(&self) -> Option<String> {
+        Some("wireObj.ptr = apiObj.share();".into())
+    }
+
+    fn wire2api_body(&self) -> String {
+        format!(
+            "return {0}.fromRaw(raw[0], raw[1], this);",
+            self.ir.dart_api_type()
+        )
+    }
+
+    fn structs(&self) -> String {
+        let field_bridge = format!(
+            "final {} bridge;",
+            self.context.config.dart_api_class_name(),
+        );
+        format!(
+            "@sealed class {0} extends FrbOpaque {{
+                {field_bridge}
+                    {0}.fromRaw(int ptr, int size, this.bridge) : super.unsafe(ptr, size);
```

Comment:
uber-nit: `{0}.fromRaw(super.ptr, super.size, this.bridge) : super.unsafe();`

Reply:
Ah not realized this part is still 2.15 (many other parts have already been upgraded). Then just keep the old way, no need for 2.17

Reply:
```
This requires the 'super-parameters' language feature to be enabled.
Try updating your pubspec.yaml to set the minimum SDK constraint to 2.17.0 or higher, and running 'pub get'
```